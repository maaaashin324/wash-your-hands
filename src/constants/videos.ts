export default Object.freeze([
  {
    title: 'PPAP-2020-/PIKOTARO(ピコ太郎)',
    subTitle: '-PIKOTARO OFFICIAL CHANNEL-',
    videoURL: 'https://www.youtube.com/watch?v=WKfolJv6Kx8',
    thumbnail: 'https://img.youtube.com/vi/WKfolJv6Kx8/hqdefault.jpg',
  },
  {
    title:
      '【Dance Video（Wash Your Hands）English Subtitled Version】〜ARASHI〜',
    subTitle: 'ARASHI',
    videoURL: 'https://www.youtube.com/watch?v=VzMT7btiwwI',
    thumbnail: 'https://img.youtube.com/vi/VzMT7btiwwI/hqdefault.jpg',
  },
  {
    title: 'Coronavirus | How to wash your hands',
    subTitle: 'Babylon Health',
    videoURL: 'https://youtu.be/seA1wbXUQTs',
    thumbnail: 'https://img.youtube.com/vi/seA1wbXUQTs/hqdefault.jpg',
  },
  {
    title: 'Hand-washing Steps Using the WHO Technique',
    subTitle: 'Johns Hopkins Medicine',
    videoURL: 'https://youtu.be/IisgnbMfKvI',
    thumbnail: 'https://img.youtube.com/vi/IisgnbMfKvI/hqdefault.jpg',
  },
]);
