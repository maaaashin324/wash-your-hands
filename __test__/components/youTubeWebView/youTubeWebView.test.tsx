import React from 'react';
import renderer from 'react-test-renderer';
import YouTubeWebView from '@/components/youTubeWebView';
import videos from '@/constants/videos';

jest.mock('react-native-webview', () => ({ WebView: 'WebView' }));

describe('YouTubeWebView', () => {
  test('should render YouTubeWebView correctly', () => {
    const tree = renderer
      .create(<YouTubeWebView uri={videos[0].videoURL} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
