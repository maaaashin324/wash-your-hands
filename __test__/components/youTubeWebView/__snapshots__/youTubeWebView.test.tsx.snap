// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouTubeWebView should render YouTubeWebView correctly 1`] = `
<WebView
  source={
    Object {
      "uri": "https://www.youtube.com/watch?v=WKfolJv6Kx8",
    }
  }
/>
`;
